admin-dev/themes/new-theme/js/pages/product/edit/specific-prices-manager.ts

```diff
@@ -0,0 +1,99 @@
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+import ProductMap from '@pages/product/product-map';
+import ProductEventMap from '@pages/product/product-event-map';
+import {EventEmitter} from 'events';
+import SpecificPriceList from '@pages/product/components/specific-price/specific-price-list';
+import Vue from 'vue';
+import VueI18n from 'vue-i18n';
+import ReplaceFormatter from '@vue/plugins/vue-i18n/replace-formatter';
+import SpecificPriceModal from '@pages/product/components/specific-price/SpecificPriceModal.vue';
+
+Vue.use(VueI18n);
+const SpecificPriceMap = ProductMap.specificPrice;
+
+export default class SpecificPricesManager {
```

Comment:
As mentioned some time ago, Im a little confused about js structure. It would be nice to have abit more domain oriented structure.
You open product/specific-price dir and you see all the related parts.
Though now it might require little more than moving this file to bring some convention back again, so I wasn't rushing for that :thinking: